include/materials/DamagePlasticityStressUpdate.h

```diff
@@ -0,0 +1,287 @@
+
+  bool requiresIsotropicTensor() override { return true; }
+
+protected:
+  virtual void initQpStatefulProperties() override;
+  virtual void finalizeReturnProcess(const RankTwoTensor & rotation_increment) override;
+
+private:
+  ///Yield function tolerance (user parameter)
+  const Real _f_tol;
+  ///dimensionless parameter involving biaxial & uniaxial compressive strengths (user parameter)
+  const Real _alfa;
+  ///dilatancy factor (user parameter)
+  const Real _alfa_p;
+  ///stiffness recovery factor (user parameter)
+  const Real _s0;
+  ///ft_ep_slope_factor_at_zero_ep (user parameter)
+  const Real _Chi;
+  ///tensile damage at half tensile strength (user parameter)
+  const Real _Dt;
+  ///yield strength in tension (user parameter)
+  const Real _ft;
+  ///fracture_energy in tension (user parameter)
+  const Real _FEt;
+  ///yield strength in compression (user parameter)
+  const Real _fyc;
+  ///compressive damage at maximum compressive strength (user parameter)
+  const Real _Dc;
+  ///maximum strength in compression (user parameter)
+  const Real _fc;
+  ///fracture energy in compression (user parameter)
+  const Real _FEc;
+
+  ///@{
+  /** The following variables are intermediate and are calculated based on the user parameters given
+   * above */
+  const Real _at;
+  const Real _ac;
+  const Real _zt;
+  const Real _zc;
+  const Real _dPhit;
+  const Real _dPhic;
+  const Real _sqrtPhit_max;
+  const Real _sqrtPhic_max;
+  const Real _dt_bt;
+  const Real _dc_bc;
+  const Real _ft0;
+  const Real _fc0;
+  ///@}
+
+  /// Intermediate variable calcualted using  user parameter tip_smoother
+  const Real _small_smoother2;
+
+  const Real _sqrt3;
+
+  /// Whether to provide an estimate of the returned stress, based on perfect plasticity
+  const bool _perfect_guess;
+
+  /// Eigenvectors of the trial stress as a RankTwoTensor, in order to rotate the returned stress back to stress space
+  RankTwoTensor _eigvecs;
+  ///damage state in tension
+  MaterialProperty<Real> & _intnl0;
+  ///damage state in compression
+  MaterialProperty<Real> & _intnl1;
+  ///element length
+  MaterialProperty<Real> & _ele_len;
+  ///fracture energy in tension
+  MaterialProperty<Real> & _gt;
+  ///fracture energy in compression
+  MaterialProperty<Real> & _gc;
+  ///tensile damage
+  MaterialProperty<Real> & _tD;
+  ///compression damage
+  MaterialProperty<Real> & _cD;
+  ///damage variable
+  MaterialProperty<Real> & _D;
+  ///minimum plastic strain
+  MaterialProperty<Real> & _min_ep;
+  ///mid value of plastic strain
+  MaterialProperty<Real> & _mid_ep;
+  ///maximum plastic strain
+  MaterialProperty<Real> & _max_ep;
+  ///damaged minimum principal stress
+  MaterialProperty<Real> & _sigma0;
+  ///damaged mid value of principal stress
+  MaterialProperty<Real> & _sigma1;
+  ///damaged maximum principal stress
+  MaterialProperty<Real> & _sigma2;
+
+  Real ft(const std::vector<Real> & intnl) const;
+  /**
```

Comment:
Put all these above the function definitions. Consider adding '@return' for these too.

Reply:
Finished